Lib/test/test_shlex.py

```diff
@@ -308,6 +308,14 @@ def testQuote(self):
             self.assertEqual(shlex.quote("test%s'name'" % u),
                              "'test%s'\"'\"'name'\"'\"''" % u)
 
+    def testJoin(self):
```

Comment:
This is a nice way to test that `join` is the inverse of `split`, but I think it may be worth a few explicit tests as well, preferably ones designed to challenge the quoting behavior.

Maybe one good test for joining would be something that seems like it could create ambiguity, like `['a ', 'b']`, `['a', ' b']` and `['a', ' ', 'b']`?

Reply:
Agreed. Adding some edge cases would be great.

Reply:
My thought was that these were already in the items used by the round-trip test, but I added these specific cases.